### ISSUE TITLE
Parse HTML files only once in the Metalsmith pipeline

### ISSUE THREAD

**Issue by cvalarida** [department-of-veterans-affairs/va.gov-team#2013]
## Problem
The current Metalsmith build has four distinct steps that each parse all the HTML files. These steps each take 6 - 10 seconds. If we parsed them only once in a discreet step, then passed that data along with the file, we can probably reduce our build time by maybe 20 seconds or more—likely a > 25% improvement.

## Goal
A faster build.

## Acceptance Criteria
- [ ] The HTML files are only parsed once

Ideally, they'd only be translated back into the buffer only after all the steps that require parsing have been completed, but we'll call that a stretch goal. Worst case scenario, we update `file.contents = new Buffer(doc.html())` at the end of every plugin.

**Comment by cvalarida** [comment]
Sample run before making any changes:

| Step | Time |
| --- | --- |
| Add `nonce` to scripts | 10.8s |
| Update external links | 6.8s |
| Add IDs to subheadings | 6.8s |
| Check for broken links | 7.2s |
| --- | --- |
| **Total** | 31.6s |

**Comment by cvalarida** [comment]
## Testing strategy
I tried generating a checksum for the files, but that doesn't work because at least one step in our Metalsmith build generates random data (adding IDs to headings...and maybe other tags?), so no two builds will be the same.

**Example:** The 404 page has a button with a randomly generated ID.
![image](https://user-images.githubusercontent.com/12970166/65567075-b4a62d80-df09-11e9-879c-07e80d0e16cd.png)

Instead, I'll have to manually check the build output to ensure that each step that parses the files still works properly:
- [ ] `nonce` is added to script tags
- [ ] `rel="noopener"` is added to links
- [ ] IDs are added to headings
- [ ] Broken links are still found

**Comment by cvalarida** [comment]
With https://github.com/department-of-veterans-affairs/vets-website/pull/10805 merged in, I'm getting the following:


| Step | Time |
| --- | --- |
| Parse HTML | 6.2s |
| Add `nonce` to scripts | 578ms |
| Update external links | 167s |
| Add IDs to subheadings | 302s |
| Check for broken links | 1.2s |
| --- | --- |
| **Total** | 8.4s |
| **Total build time for all steps** | 40.7s |
